src/lib/swapper/swappers/ThorchainSwapper/evm/routerCalldata.ts

```diff
@@ -1,18 +1,18 @@
-import Web3 from 'web3'
+import { encodeFunctionData } from 'viem'
 import { routerAbi } from 'lib/swapper/swappers/ThorchainSwapper/evm/routerAbi'
 
 export const depositWithExpiry = (
-  contractAddress: string,
+  _contractAddress: string,
   vault: string,
   asset: string,
   amount: string,
   memo: string,
   expiry: number,
 ) => {
-  const web3 = new Web3()
-  const routerContract = new web3.eth.Contract(routerAbi, contractAddress)
-  const data = routerContract.methods
-    .depositWithExpiry(vault, asset, amount, memo, expiry)
-    .encodeABI()
+  const data = encodeFunctionData({
+    abi: routerAbi,
+    functionName: 'depositWithExpiry',
+    args: [vault, asset, amount, memo, expiry],
+  })
```

Comment:
viem used instead of Web3 here, as the commonjs build of `web3-eth-contract` is turbo rugged, meaning the jest tests will fail at runtime.
@0xApotheosis lost sanity on that one and so di I, spent hours debugging the commonjs flavor of the module internally, before ragequitting and switching to viem, which would be a lighter and just as type-safe choice here.